### PR TITLE
[B-37655]:Added obligations data to context

### PR DESCRIPTION
**Pull request by singhs1**
Added obligations data to context (if present)

### CODE REVIEW

File: grpc_opa/authorizer.go [hunk 1]
```diff
@@ -240,7 +240,10 @@ func (a *DefaultAuthorizer) Evaluate(ctx context.Context, fullMethod string, opa
 			trace.StringAttribute("out", string(raw)),
 		}, "out")
 	}
-
+	// adding obligations data to context if present
+	if ob,ok := response["obligations"].([]string);ok{
```
Comment (drewwells) [inline grpc_opa/authorizer.go:244]: can you run this through go fmt? Apparently we don't check for that on build right now.

File: grpc_opa/authorizer.go [hunk 2]
```diff
@@ -240,7 +240,10 @@ func (a *DefaultAuthorizer) Evaluate(ctx context.Context, fullMethod string, opa
...
+		ctx = context.WithValue(ctx,"obligations",ob)
```
Comment (drewwells) [inline grpc_opa/authorizer.go:245]: change "obligations" to a private type. See this as an example https://github.com/Infoblox-CTO/atlas.eventing/pull/31/files#diff-c553b059ae0d06bffc1383e72286221924a201e5dc96029dc3b4c9174e46ad93R56

Comment (drewwells) [reply grpc_opa/authorizer.go:245]: The struct{} usage here is a bit unique. An approach I see more often is this
```
type key obligationKey = "obligations"

ctx := context.WithValue(key, ....)
```

**Review by drewwells — APPROVED** [review]
lgtm

**Review by rchowinfoblox — CHANGES_REQUESTED** [review]
Shouldn't we capitalize obligationKey and obKey so they're exported to allow other apps to use?

**Comment by singhs1** [comment]
> Shouldn't we capitalize obligationKey and obKey so they're exported to allow other apps to use?

Done

**Review by rchowinfoblox — APPROVED** [review]

**Pull request merged**